### PR TITLE
refactor(compute): create Images samples

### PR DESCRIPTION
**Pull request by ssvir**
implemented samples : compute_images_create_from_image, compute_images_create_from_snapshot, compute_images_get,compute_images_get_from_family,compute_images_get_list,compute_images_set_deprecation_status

## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/images/CreateFromImage.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Google Cloud project you want to use.
+    String projectId = "your-project-id";
+    // Name of the image you want to copy.
+    String sourceImageName = "your-image-name";
+    // Name of the image you want to create.
+    String imageName = "your-image-name";
+    // Name of the project that hosts the source image. If left unset, it's assumed to equal
+    // the `project_id`.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateFromImage.java:44]: Should this be `projectId`?

File: compute/cloud-client/src/main/java/compute/images/CreateFromImage.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    String sourceProjectId = "your-source-project-id";
+    // An iterable collection of guest features you want to enable for the bootable image.
+    // Learn more about Guest OS features here:
+    // https://cloud.google.com/compute/docs/images/
+    // create-delete-deprecate-private-images#guest-os-features
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateFromImage.java:49]: Links should be in one line.

File: compute/cloud-client/src/main/java/compute/images/CreateFromImage.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    // `us` multi-region, or "us-central1" to store it in the `us-central1` region.
+    // If you do not make a selection,
+    // Compute Engine stores the image in the multi-region closest to your image's source location.
+    String storageLocation = "your-storage-location";
+
+    createImageFromImage(projectId, sourceImageName, imageName,
+            sourceProjectId, guestOsFeature, storageLocation);
+  }
+
+  // Creates a new disk image.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateFromImage.java:61]: The crux is creating an image from an existing one. Update the description to reflect that.

File: compute/cloud-client/src/main/java/compute/images/CreateFromImage.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      Image.Builder imageResource = Image.newBuilder()
+              .setName(imageName)
+              .setSourceImage(sourceImage.getSelfLink());
+
+      if (storageLocation != null) {
+        imageResource.addStorageLocations(storageLocation);
+      }
+      if (guestOsFeature != null) {
+        for (String feature : guestOsFeature) {
+          GuestOsFeature build = GuestOsFeature.newBuilder()
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateFromImage.java:83]: Update the variable name to be more apt. Something like `guestOsFeature`

File: compute/cloud-client/src/main/java/compute/images/CreateFromSnapshot.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import com.google.cloud.compute.v1.SnapshotsClient;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateFromSnapshot {
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateFromSnapshot.java:35]: CreateImageFromSnapshot?

File: compute/cloud-client/src/main/java/compute/images/CreateFromSnapshot.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    // https://cloud.google.com/compute/docs/images/
+    // create-delete-deprecate-private-images#guest-os-features
+    List<String> guestOsFeature = new ArrayList<>();
+    // The storage location of your image. For example, specify "us" to store the image in the
+    // `us` multi-region, or "us-central1" to store it in the `us-central1` region.
+    // If you do not make a selection,
+    // Compute Engine stores the image in the multi-region closest to your image's source location.
+    String storageLocation = "your-storage-location";
+
+    createImageFromImage(projectId, sourceSnapshotName, imageName,
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateFromSnapshot.java:59]: Update the method name.

File: compute/cloud-client/src/main/java/compute/images/CreateFromSnapshot.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            sourceProjectId, guestOsFeature, storageLocation);
+  }
+
+  // Creates an image based on a snapshot.
+  public static Image createImageFromImage(String projectId, String sourceSnapshotName,
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateFromSnapshot.java:64]: Update the method name `createImageFromSnapshot`

File: compute/cloud-client/src/main/java/compute/images/GetImage.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+import com.google.cloud.compute.v1.Image;
+import com.google.cloud.compute.v1.ImagesClient;
+import java.io.IOException;
+
+public class GetImage {
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Google Cloud project you want to use.
+    String projectId = "your-project-id";
+    // Name of the image you want to create.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/GetImage.java:31]: Update the description.

File: compute/cloud-client/src/main/java/compute/images/CreateFromImage.java [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+      }
+
+      InsertImageRequest request = InsertImageRequest.newBuilder()
+              .setProject(projectId)
+              .setRequestId(UUID.randomUUID().toString())
+              .setImageResource(imageResource)
+              .build();
+      client.insertCallable().futureCall(request).get(60, TimeUnit.SECONDS);
+
+      return client.get(projectId, imageName);
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateFromImage.java:97]: Print the value before returning. Applies to all samples.

File: compute/cloud-client/src/main/java/compute/images/GetImageFromFamily.java [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+import com.google.cloud.compute.v1.Image;
+import com.google.cloud.compute.v1.ImagesClient;
+import java.io.IOException;
+
+public class GetImageFromFamily {
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Google Cloud project you want to use.
+    String projectId = "your-project-id";
+    // Name of the image family you want to create.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/GetImageFromFamily.java:31]: Update the description.
`Name of the image family you want to retrieve the image from`?

File: compute/cloud-client/src/main/java/compute/images/GetImageFromFamily.java [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    String family = "your-family-name";
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/GetImageFromFamily.java:32]: Specify an actual family name.

File: compute/cloud-client/src/main/java/compute/images/GetImageFromFamily.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (ImagesClient client = ImagesClient.create()) {
+      GetFromFamilyImageRequest request = GetFromFamilyImageRequest.newBuilder()
+              .setProject(projectId)
+              .setFamily(family)
+              .build();
+
+      // List of public operating system (OS) images:
+      // https://cloud.google.com/compute/docs/images/os-details
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/GetImageFromFamily.java:48]: Move this comment to main.

File: compute/cloud-client/src/main/java/compute/images/SetImageDeprecationStatus.java [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class SetImageDeprecationStatus {
+  public static void main(String[] args)
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Google Cloud project you want to use.
+    String projectId = "your-project-id";
+    // Name of the image you want to create.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/SetImageDeprecationStatus.java:37]: Update description.

File: compute/cloud-client/src/main/java/compute/images/SetImageDeprecationStatus.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+    String imageName = "your-image-name";
+    // Name of the image you want to create.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/SetImageDeprecationStatus.java:39]: Update description.

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/test/java/compute/images/ImagesIT.java [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+    Assert.assertNotNull(images);
+    Assert.assertTrue(images.size() > 1);
+    Assert.assertTrue(images.stream().anyMatch(image
+            -> image.getName().equals(IMAGE_FROM_IMAGE_NAME)));
+  }
+
+  @Test
+  public void stage4_setImageDeprecationStatus()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    Thread.sleep(100000);
```
[the remark this answers is not in the document]
Comment (ssvir) [reply compute/cloud-client/src/test/java/compute/images/ImagesIT.java:154]: cause to set status, your image should be in a running state. In previous tests, you can easily retrieve info about the image but to change the status you should wait till the image is fully run

File: compute/cloud-client/src/main/java/compute/images/CreateImageFromImage.java [hunk 16]
```diff
@@ -58,10 +57,10 @@ public static void main(String[] args)
             sourceProjectId, guestOsFeature, storageLocation);
   }
 
-  // Creates a new disk image.
+  // Creates a copy of another image.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateImageFromImage.java:60]:
```suggestion
  // Creates a new disk image from an existing image.
```

File: compute/cloud-client/src/main/java/compute/images/CreateImageFromImage.java [hunk 17]
```diff
@@ -78,12 +77,12 @@ public static Image createImageFromImage(String projectId, String sourceImageNam
...
       }
-      if (guestOsFeature != null) {
-        for (String feature : guestOsFeature) {
-          GuestOsFeature build = GuestOsFeature.newBuilder()
-                  .setType(feature)
-                  .build();
-          imageResource.addGuestOsFeatures(build);
+      if (guestOsFeatures != null) {
+        for (String feature : guestOsFeatures) {
+          GuestOsFeature.Builder guestOsFeature = GuestOsFeature.newBuilder()
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateImageFromImage.java:82]:
```suggestion
          GuestOsFeature.Builder guestOsFeatureBuilder = GuestOsFeature.newBuilder()
```

File: compute/cloud-client/src/main/java/compute/images/CreateImageFromImage.java [hunk 18]
```diff
@@ -78,12 +77,12 @@ public static Image createImageFromImage(String projectId, String sourceImageNam
...
+                  .setType(feature);
+
+          imageResource.addGuestOsFeatures(guestOsFeature);
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/images/CreateImageFromImage.java:85]:
```suggestion
          imageResource.addGuestOsFeatures(guestOsFeatureBuilder);
```

File: compute/cloud-client/src/main/java/compute/images/CreateImageFromImage.java (hunk 17 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/images/CreateImageFromImage.java:82]: It's confusing to have `guestOsFeatures` and `guestOsFeature` denote different data types. Update this across snippets.

**Review by Sita04 — CHANGES_REQUESTED** [review]
Kindly address the test failures and the comments.

**Review by Sita04 — APPROVED** [review]
LGTM. Please merge once tests pass.

**Pull request merged**